signature/jws/signer.go

```diff
@@ -86,6 +87,26 @@ func NewSignerWithCertificateChain(method jwt.SigningMethod, key crypto.PrivateK
 	}, nil
 }
 
+// NewSignerFromFiles creates a signer from a key and certificate files.
+// Since the provided certificate signing key could potentially be a remote key, the relation of the
+// signing key and its certificate chain is not verified, and should be verified by the caller.
+func NewSignerFromCertificate(cert tls.Certificate) (*Signer, error) {
+	certs := make([]*x509.Certificate, len(cert.Certificate))
+	for i, c := range cert.Certificate {
+		cert, err := x509.ParseCertificate(c)
+		if err != nil {
+			return nil, err
+		}
+		certs[i] = cert
+	}
+	key := cert.PrivateKey
+	method, err := SigningMethodFromKey(key)
+	if err != nil {
+		return nil, err
+	}
+	return NewSignerWithCertificateChain(method, key, certs)
```

Comment:
This call is going to fail if `len(certs) == 0`. This isn't wrong but seems like a lot of work above is wasted in this case. Did you consider verifying `len(certs) > 0` at the start of this method?

Reply:
> Did you consider verifying `len(certs) > 0` at the start of this method?

I didn't, but now I will 😄